src/interface.js

```diff
@@ -0,0 +1,16 @@
+const client = new Client();
+
+function makeHeadlinesRequest(callback) {
+  client.fetchHeadlines().then(callback);
+}
+
+function makeSummaryRequest(url, callback) {
+  client.fetchSummary(url).then(callback);
+}
+
+function makeArticleRequest(url, callback) {
+  client.fetchArticle(url).then(callback);
+}
+
+const app = new App(makeHeadlinesRequest, makeSummaryRequest, makeArticleRequest);
+window.addEventListener('load', () => app.getHeadlines());
```

Comment:
Again, it feels like your well planned and named methods in your other classes make this very easy to read. I think it shows a good understanding of encapsulation to keep this so readable and simple.